contracts/roles/RevocationRegistryOnChain.sol

```diff
@@ -0,0 +1,97 @@
+// SPDX-License-Identifier: GPL-3.0
+
+pragma solidity >= 0.7.0 < 0.9.0;
+
+import "@ensdomains/ens-contracts/contracts/registry/ENSRegistry.sol";
+import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import "@openzeppelin/contracts/utils/introspection/ERC165Checker.sol";
+import "@ew-did-registry/proxyidentity/contracts/IOwned.sol";
+import "./RoleDefinitionResolver.sol";
+import {ClaimManager} from "./ClaimManager.sol";
+
+interface EthereumDIDRegistry {
+    function identityOwner(address identity) external view returns(address);
+    function validDelegate(address identity, bytes32 delegateType, address delegate) external view returns(bool);
+}
+    
+contract RevocationRegistryOnChain {
+    bytes32 constant private ASSERTION_DELEGATE_TYPE = 0x766572694b657900000000000000000000000000000000000000000000000000;
+    string constant private ERC712_DOMAIN_NAME = "Revocation Registry";
+    string constant private ERC712_DOMAIN_VERSION = "1.0";
+
+    struct RevokedClaim {
+        address revoker;
+        uint revokedTimestamp;
+        bytes32 revokerRoleDigest; //namehash
+    }
+
+    mapping(bytes32 => RevokedClaim) revokedClaims;
```

Comment:
What is the rationale for the `bytes32` index into the map? I think an alternative that might be better is
`mapping(bytes32 => mapping(address => RevokedClaim)) private revokedClaims;`, where `bytes32` is the role namehash and `address` is the ethr DID.
My thinking:
- It is a more intuitive API. Currently, one need to do know to take the 
- If we ever needed to migrate the contract, depending on what we need to change, it might be useful to know which DID+role a particular revocation refers to. 
- I guess the current approach might be a little more privacy preserving as one can't enumerate the revoked claims of a DID. But I think this doesn't have much benefit it would be trivial to hash all of the `role+addr` combinations for a given DID and check them all.

Reply:
`bytes32` index makes it easy to refer the revocation without knowing the revokers address. This makes the validation of revocation convenient in the sense that only `bytes32`(namehash of role and subject's address) is needed.

`mapping(bytes32 => mapping(address => RevokedClaim)) private revokedClaims;` this would create a need to store revokers and revokedClaim to be stored and mapped somewhere outside the contract as well, in order to check the status of the claim. 

Could you please elaborate more on the 2nd and 3rd point? I didn't understand that.

Reply:
I think that maybe we're not on the same page. So with `mapping(bytes32 => mapping(address => RevokedClaim)) private revokedClaims;`, `bytes32` is the `namehash` of the role namespace, `address` is the DID of the `subject`, not the `revoker`. So one could check if a claim is revoked by checking `revokedClaims[namehash(role)][subjectAddr]`.

Reply:
@jrhender Gotcha, now we are on same page. Will make the suggested change.